libgstc/pygstc/gstc.py

```diff
@@ -0,0 +1,788 @@
+
+        Returns
+        -------
+        result : bool
+            Whether or not GSTD is running in localhost
+        """
+
+        if self._ip not in ['localhost', '127.0.0.1']:
+
+            # bypass process check, we don't know how to start gstd remotely
+
+            self._logger.warning(
+                'Assuming GSTD is running in the remote host at %s' %
+                self._ip)
+            return True
+        for proc in psutil.process_iter():
+
+            # check whether the process name matches
+
+            if proc.name() == GSTD_PROCNAME:
+                return True
+            else:
+                self._logger.error('GStreamer Daemon is not running')
+                return False
+
+    def bus_filter(self, pipe_name, filter):
+        """
+        Select the types of message to be read from the bus. Separate
+        with a '+', i.e.: eos+warning+error.
+
+        Parameters
+        ----------
+        pipe_name: string
+            The name of the pipeline
+        filter: string
+            Filter to be applied to the bus. '+' reparated strings
+        """
+
+        self._logger.info('Setting bus read filter of pipeline %s to %s'
+                          % (pipe_name, filter))
+        self._send_cmd_line(['bus_filter', pipe_name, filter])
+
+    def bus_read(self, pipe_name):
+        """
+        Read the bus and wait.
+
+        Parameters
+        ----------
+        pipe_name: string
+            The name of the pipeline
+
+        Returns
+        -------
+        result : dictionary
+            Command response
+        """
+
+        self._logger.info('Reading bus of pipeline %s' % pipe_name)
+        result = self._send_cmd_line(['bus_read', pipe_name])
+        return result['response']
+
+    def bus_timeout(self, pipe_name, timeout):
+        """
+        Apply a timeout for the bus polling.
+        Parameters
+        ----------
+        pipe_name: string
+            The name of the pipeline
+        timeout: string
+            Timeout in nanosecons. -1: forever, 0: return
+            immediately, n: wait n nanoseconds.
+        """
+
+        self._logger.info('Setting bus read timeout of pipeline %s to %s'
+                          % (pipe_name, timeout))
+        self._send_cmd_line(['bus_timeout', pipe_name, timeout])
+
+    def create(
+        self,
+        uri,
+        property,
+        value,
+    ):
+        """
+        Creates a resource at the given URI.
+
+        Parameters
+        ----------
+        uri: string
+            Resource identifier
+        property: string
+            The name of the property
+        value: string
+            The initial value to be set
+        """
+
+        self._logger.info('Creating property %s in uri %s with value "%s"'
+                          % (property, uri, value))
+        self._send_cmd_line(['create', uri, property, value])
+
+    def debug_color(self, colors):
```

Comment:
It is strange to receive a string boolean such as `'true'` or `'false'`. From the python binding we should receive a native boolean (`True` and `False`) and do the conversion ourselves.

Reply:
Please apply this change to all booleans.